File: src/main/java/net/sharplab/epubtranslator/core/provider/epub/EPubChapterProvider.java
package net.sharplab.epubtranslator.core.provider.epub;

import net.sharplab.epubtranslator.core.model.EPubChapter;
import net.sharplab.epubtranslator.core.model.FileEntry;

/**
 * EPubChapterを生成するプロバイダ
 */
public class EPubChapterProvider implements EPubContentFileProvider {

    @Override
    public boolean canHandle(FileEntry fileEntry) {
        return fileEntry.getName().endsWith(".xhtml") || fileEntry.getName().endsWith(".html")  || fileEntry.getName().endsWith(".htm");
    }

    @Override
    public EPubChapter provide(FileEntry fileEntry) {
        if(!canHandle(fileEntry)){
            throw new IllegalArgumentException();
        }
        return new EPubChapter(fileEntry.getName(), fileEntry.getData());
    }
}

File: src/main/java/net/sharplab/epubtranslator/core/provider/epub/EPubChapterProvider.kt
package net.sharplab.epubtranslator.core.provider.epub

import net.sharplab.epubtranslator.core.model.EPubChapter
import net.sharplab.epubtranslator.core.model.FileEntry

/**
 * EPubChapterを生成するプロバイダ
 */
class EPubChapterProvider : EPubContentFileProvider {
    override fun canHandle(fileEntry: FileEntry): Boolean {
        return fileEntry.name.endsWith(".xhtml") || fileEntry.name.endsWith(".html") || fileEntry.name.endsWith(".htm")
    }

    override fun provide(fileEntry: FileEntry): EPubChapter {
        require(canHandle(fileEntry))
        return EPubChapter(fileEntry.name, fileEntry.data)
    }
}
